Rollbar/RollbarLogger.cs

```diff
@@ -191,66 +209,104 @@ public ILogger Debug(object obj, IDictionary<string, object> custom = null)
         {
-            return SendBody(new Body(e), level, custom);
+            SendBodyAsync(new Body(new Message(message)), level, custom);
         }
 
-        private Guid? Report(string message, ErrorLevel? level = ErrorLevel.Error, IDictionary<string, object> custom = null)
+        private void SendBodyAsync(Body body, ErrorLevel? level, IDictionary<string, object> custom)
         {
-            return SendBody(new Body(new Message(message)), level, custom);
+            // we are taking here a fire-and-forget approach:
+            Task.Factory.StartNew(() => SendBody(body, level, custom));
         }
 
-        private Guid? SendBody(Body body, ErrorLevel? level, IDictionary<string, object> custom)
+        private void SendBody(Body body, ErrorLevel? level, IDictionary<string, object> custom)
         {
-            if (string.IsNullOrWhiteSpace(this._config.AccessToken) 
-                || this._config.Enabled == false
-                )
+            lock(this._syncRoot)
             {
-                return null;
-            }
+                if (string.IsNullOrWhiteSpace(this._config.AccessToken)
+                    || this._config.Enabled == false
+                    )
+                {
+                    return;
+                }
 
-            var guid = Guid.NewGuid();
+                var data = new Data(this._config.Environment, body)
+                {
+                    Custom = custom,
+                    Level = level ?? this._config.LogLevel
+                };
 
-            var data = new Data(this._config.Environment, body)
-            {
-                Custom = custom,
-                Level = level ?? this._config.LogLevel
-            };
+                var payload = new Payload(this._config.AccessToken, data);
+                payload.Data.GuidUuid = Guid.NewGuid();
+                payload.Data.Person = this._config.Person;
 
-            var payload = new Payload(this._config.AccessToken, data);
-            payload.Data.GuidUuid = guid;
-            payload.Data.Person = this._config.Person;
+                if (this._config.Server != null)
+                {
+                    payload.Data.Server = this._config.Server;
+                }
 
-            if (this._config.Server != null)
-            {
-                payload.Data.Server = this._config.Server;
-            }
+                try
+                {
+                    if (this._config.CheckIgnore != null
+                        && this._config.CheckIgnore.Invoke(payload)
+                        )
+                    {
+                        return;
+                    }
+                }
+                catch (System.Exception ex)
+                {
+                    OnRollbarEvent(new InternalErrorEventArgs(this._config, payload, ex, "While  check-ignoring a payload..."));
+                    return;
```

Comment:
If the check ignore function raises an exception, we still want to send the payload, this looks like we return early rather than sending it.

Reply:
corrected